### ISSUE TITLE
"memory" filesystem throws FileNotFoundError when deleting multilevel folders

### ISSUE THREAD

**Issue by namanjain** [intake/filesystem_spec#449]
When you try to delete a dir recursively in "memory" filesystem, it fails if the directory has only one sub-directory with files in it. So something like this breaks:

```
>>> import fsspec
>>> fs = fsspec.filesystem("memory")
>>> with fs.open("/foo/bar/baz", "w") as f:
...   f.write("qux")
...
3
>>> fs.rm("/foo", recursive=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/naman/.pyenv/versions/bionic/lib/python3.8/site-packages/fsspec/spec.py", line 821, in rm
    self.rm_file(p)
  File "/Users/naman/.pyenv/versions/bionic/lib/python3.8/site-packages/fsspec/spec.py", line 797, in rm_file
    self._rm(path)
  File "/Users/naman/.pyenv/versions/bionic/lib/python3.8/site-packages/fsspec/implementations/memory.py", line 165, in _rm
    raise FileNotFoundError
FileNotFoundError
```

It breaks even if I specify the `maxdepth`, although as per the [rm documentation](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem.rm), no depth means no limit.

FWIW, the same structure can be deleted in other filesystems, like "file" filesystem.

**Comment by namanjain** [comment]
I added some print statements to see what's going on. It first deletes `/foo/bar/baz` and then, as you suspected, fails when trying to delete the directory `foo/bar/`.

But it works when you don't have the `bar` directory and `baz` is directly inside `foo`, basically `/foo/baz`. In that case, it only deletes `/foo/baz` file and doesn't try to delete the `/foo/` directory.

**Comment by martindurant** [comment]
It sounds like you have a nice test case that can be added to the fsspec/implementations/test_memory.py file, and a potential fix which ignores errors from rmdir (see the attribute `pseudo_dirs`) either during recursive `rm` or perhaps generally.

**Comment by namanjain** [comment]
@martindurant I have a proposed fix for this bug but I don't seem to have required access. How do I request write access to the repo? Would you be able to grant it?

**Comment by martindurant** [comment]
The standard procedure is to
- fork the repo (button in the very top right of this page), giving you a version where you do have write permission under your own account
- add a new remote to your local repo pointing to the new fork
- push a branch to your version of the repo
- submit a pull request, either from the github screen showing your repo, or by the URL that git will give you when pushing your changes.

For safety reasons, we keep the number of people with write permissions to the repo as small as possible.

**Comment by namanjain** [comment]
This was fixed in #454.
